redhat-appstudio/remote-secret
# Distinguish between data and stringData in RS data upload

### What does this PR do?
This changes the type of `data` field of the remote secret from `map[string]string` to `map[string][]byte` and adds another field, `stringData` with the type `map[string]string`. This is to behave in the same way as `data` and `stringData` of the ordinary Kubernetes secret are defined.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-618

### How to test this PR?
1. `kubectl apply -f samples/remote-secret-with-data.yaml`
2. `kubectl patch remotesecret myrs1 --type=merge -p='{"spec": {"targets": [{"namespace": "default"}]}}'
3. `SECRET_NAME=$(kubectl get remotesecret myrs1 -o json | jq -r '.status.targets[0].secretName')`
4. `kubectl get secret $SECRET_NAME -o json | jq -r '.data.username' | base64 -d` should give you "gena" (which is the actual value) instead of its base64 representation.
